app/invoices/payment/controller.js

```diff
@@ -3,11 +3,23 @@ import Ember from 'ember';
 import PatientSubmodule from 'hospitalrun/mixins/patient-submodule';
 
 export default AbstractEditController.extend(PatientSubmodule, {
+  closeModalAction: 'cleanup',
   cancelAction: 'closeModal',
   findPatientVisits: false,
   invoiceController: Ember.inject.controller('invoices'),
   newPayment: false,
 
+  actions: {
+    cleanup: function() {
+      var model = this.model;
+      this.model.validate().then(function() {
+        model.save();
```

Comment:
This cleanup code should not save the model because the model should not be saved on the cancel action, it should only be saved when the user clicks on the update/add action which is already handled elsewhere in the controller.